server/request.go

```diff
@@ -10,14 +10,21 @@ import (
 type Request interface {
 	context.Context
 
-	Body() []byte
+	// Payload are the actual bytes returned from the transport
+	Payload() []byte
+
+	// Body is the Unmarshalled `Payload()`. If `RequestType()` is set on
+	// the `Endpoint`, we can attempt to unmarshal it for you
```

Comment:
Presumably this would mean an end user should use `Payload()` rather than `Body()` if they haven't set the `RequestType`?